lib/internal/test_runner/test.js

```diff
@@ -149,14 +150,23 @@ class Test extends AsyncResource {
       this.timeout = parent.timeout;
     }
 
-    if (isUint32(concurrency) && concurrency !== 0) {
-      this.concurrency = concurrency;
-    } else if (typeof concurrency === 'boolean') {
-      if (concurrency) {
-        this.concurrency = isTestRunner ? MathMax(cpus().length - 1, 1) : Infinity;
-      } else {
-        this.concurrency = 1;
-      }
+    switch (typeof concurrency) {
+      case 'number':
+        validateUint32(concurrency, 'concurrency', 1);
+        this.concurrency = concurrency;
+        break;
+
+      case 'boolean':
+        if (concurrency) {
+          this.concurrency = isTestRunner ? MathMax(cpus().length - 1, 1) : Infinity;
+        } else {
+          this.concurrency = 1;
+        }
+        break;
+
+      default:
+        if (concurrency != null)
+          validateOneOf(concurrency, 'concurrency', ['boolean', 'number']);
```

Comment:
```suggestion
          validateOneOf(typeof concurrency, 'concurrency', ['boolean', 'number']);
```
Since the `validateOneOf()` validator directly compares the value with the elements in the array, not their type.

Also I think it's a better idea to validate this before the `switch` statement instead of comparing the types again after going through them in the `switch` `case`s, only if `concurrency` is not nullsih of course.

Something like:

```js
if (concurrency != null) {
  const concurrencyType = typeof concurrrency;

  validateOneOf(concurrencyType, 'concurrency', ['boolean', 'number']);

  switch (concurrencyType) { ... }
}
```